### ISSUE TITLE
DAList.add_action() does not ask for object type when .ask_object_type = True

### ISSUE THREAD

**Issue by Gauntlet173** [jhpyle/docassemble#132]
Noticed this while testing an app, yesterday. Took the demo from the documentation and modified it to demonstrate the problem. When you get to the review screen, and click on the add action button, it does not ask for the new_object_type, but just adds the same object type that was most recently added.

```
objects:
  - location: DAList
---
mandatory: True
code: |
  location.there_are_any = True
  location.ask_object_type = True
---
mandatory: True
question: |
  The locations
subquestion: |
  % for loc in location:
  * ${ loc }
  % endfor
field: locations_seen
---
question: |
  Do you know the full address of the
  ${ ordinal(location.current_index()) }
  location?
buttons:
  - Yes:
      code: |
        location.new_object_type = Address
  - No:
      code: |
        location.new_object_type = City
---
question: |
  What is the address of the
  ${ ordinal(i) } location?
fields:
  - Address: location[i].address
  - Unit: location[i].unit
    required: False
  - City: location[i].city
  - State: location[i].state
    code: |
      states_list()
  - Zip: location[i].zip
    required: False
---
question: |
  What is the city of the
  ${ ordinal(i) } location?
fields:
  - City: location[i].city
  - State: location[i].state
    code: |
      states_list()
---
question: |
  Would you like to add another location?
yesno: location.there_is_another
---
table: location.table
rows: location
columns:
  - City: row_item.city
---
mandatory: True
field: location_review
question: Review Locations
subquestion: |
 ${ location.table }
 
 ${ location.add_action() }
---
```

**Comment by jhpyle** [comment]
I think I have fixed this in version 0.4.20.

**Comment by jhpyle** [comment]
Closing this for now but feel free to reopen if `add_action()` misbehaves in the future.
